### PR TITLE
Change float64 -> float32

### PR DESCRIPTION
**Pull request by AltamashRafiq**
Python's XGBoost uses float32 internally instead of float64. This PR updates this repo to reflect this behavior.

### CODE REVIEW

**Comment by AltamashRafiq** [comment]
@Elvenson @jcronq here is the float64 -> float32 change we have been discussing. A test that fails with float64 but passes with float32 is added.

**Review by Elvenson — COMMENTED** [review]
Hi @AltamashRafiq, thank you for your amazing contribution! I really appreciate it. I just have some small comments. Can you help check? Thanks.

File: mat/mat.go [hunk 1]
```diff
@@ -74,15 +74,15 @@ func ReadLibsvmFileToSparseMatrix(fileName string) (SparseMatrix, error) {
 			if err != nil {
 				return SparseMatrix{}, fmt.Errorf("cannot parse to float %s: %s", pair[1], err)
 			}
-			vec[int(colIdx)] = val
+			vec[int(colIdx)] = float32(val)
```
[the remark this answers is not in the document]
Comment (AltamashRafiq) [reply mat/mat.go:77]: Thanks! Just put in the fix.

File: mat/mat.go [hunk 2]
```diff
@@ -106,15 +106,15 @@ func ReadCSVFileToDenseMatrix(fileName string, delimiter string, defaultVal floa
 		tokens := strings.Split(line, delimiter)
 		vec := Vector{}
 		for i := 0; i < len(tokens); i++ {
-			var val float64
+			var val float32
 			if len(tokens[i]) == 0 {
 				val = defaultVal
 			} else {
 				v, err := strconv.ParseFloat(tokens[i], 64)
```
[the remark this answers is not in the document]
Comment (AltamashRafiq) [reply mat/mat.go]: Done.

**Review by Elvenson — APPROVED** [review]

**Pull request merged**